daemon/list.go

```diff
@@ -90,6 +90,41 @@ func (daemon *Daemon) Containers(config *types.ContainerListOptions) ([]*types.C
 	return daemon.reduceContainers(config, daemon.transformContainer)
 }
 
+func (daemon *Daemon) filterByNameIDMatches(ctx *listContext) []*container.Container {
+	names := ctx.filters.Get("name")
+	ids := ctx.filters.Get("id")
+	if len(names)+len(ids) == 0 {
+		// if name or ID filters are not in use, return to
+		// standard behavior of walking the entire container
+		// list from the daemon's in-memory store
+		return daemon.List()
+	}
+
+	matches := make(map[string]bool)
+	// look for name matches; errors only report not found and can be ignored
+	if len(names) > 0 {
+		for id, idNames := range ctx.names {
+			for _, eachName := range idNames {
```

Comment:
Wondering; should we filter by fullId first (the other filter below) and, if the `id` is already matched, skip trying to match by name? I can imagine (haven't checked) that matching by full ID is faster then matching by partial name, and there's no reason to match the same ID twice